lib/driverProviders/driverProvider.ts

```diff
@@ -89,42 +88,38 @@ export abstract class DriverProvider {
 
 
   /**
-   * Quits an array of drivers and returns a q promise instead of a webdriver one
+   * Quits an array of drivers and returns a native promise instead of a webdriver one
    *
    * @param drivers {webdriver.WebDriver[]} The webdriver instances
    */
-  static quitDrivers(provider: DriverProvider, drivers: WebDriver[]): q.Promise<void> {
-    let deferred = q.defer<void>();
-    wdpromise
-        .all(drivers.map((driver: WebDriver) => {
-          return provider.quitDriver(driver);
-        }))
-        .then(
-            () => {
-              deferred.resolve();
-            },
-            () => {
-              deferred.resolve();
-            });
-    return deferred.promise;
+  static quitDrivers(provider: DriverProvider, drivers: WebDriver[]): Promise<void> {
+    let deferredResolve: (x?: any) => void;
+    let deferredReject: (x?: any) => void;
+    let deferred = new Promise<void>((resolve, reject) => {
+      deferredResolve = resolve;
+      deferredReject = reject;
+    });
+    wdpromise.all(drivers.map((driver: WebDriver) => provider.quitDriver(driver)))
+        .then(() => deferredResolve(), () => deferredReject());
```

Comment:
Could this just be

```
return wdpromise.all(drivers.map((driver: WebDriver) => provider.quitDriver(driver)))
```

Since it will `then` or `catch` further up the promise chain?

If we have to return a native promise, how about just 

```
return new Promise((resolve, reject) => {
  wdpromise.all(drivers.map((driver: WebDriver) => provider.quitDriver(driver)))
    .then(resolve, reject)
})
```